fgpyo/io/__init__.py

```diff
@@ -56,11 +56,15 @@
 from typing import Iterator
 from typing import Set
 from typing import TextIO
+from typing import TypeAlias
 from typing import Union
 from typing import cast
 
 COMPRESSED_FILE_EXTENSIONS: Set[str] = {".gz", ".bgz"}
 
+ReadableFileHandle: TypeAlias = Union[io.TextIOWrapper, TextIO, IO[Any]]
+WritableFileHandle: TypeAlias = Union[IO[Any], io.TextIOWrapper]
```

Comment:
I _**think**_ you only need `io.TextIOWrapper` https://github.com/fulcrumgenomics/fgpyo/pull/103#discussion_r1539758711

But I also recently found these structural types in typeshed too:

https://github.com/python/typeshed/blob/2d33fe212221a05661c0db5215a91cf3d7b7f072/stdlib/_typeshed/__init__.pyi#L253-L271

Reply:
1. It'd be nice to have comment or two about what each of these types are trying to alias
2. Why not just `Readable` and `Writeable`?